hashring/hashring.go

```diff
@@ -215,53 +215,43 @@ func (r *HashRing) ServerCount() int {
 	return count
 }
 
+// Lookup returns the owner of the given key.
 func (r *HashRing) Lookup(key string) (string, bool) {
-	r.servers.RLock()
-
-	iter := r.servers.tree.IterAt(int(r.hashfunc([]byte(key))))
-	if iter.Nil() {
-		r.servers.RUnlock()
+	strs := r.LookupN(key, 1)
```

Comment:
Actually, wait, I'm not sure I do.  If the key isn't present in the map, won't we still get a result this way?